classes/security/authorization/internal/SubmissionFileRevisionStageRequiredPolicy.inc.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * @file classes/security/authorization/internal/SubmissionFileRevisionStageRequiredPolicy.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2000-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionFileRevisionStageRequiredPolicy
+ * @ingroup security_authorization_internal
+ *
+ * @brief Submission file policy to ensure that we have a file at revision stage.
+ *
+ */
+
+import('lib.pkp.classes.security.authorization.internal.SubmissionFileBaseAccessPolicy');
+
+class SubmissionFileRevisionStageRequiredPolicy extends SubmissionFileBaseAccessPolicy {
+	/**
+	 * Constructor
+	 * @param $request PKPRequest
+	 */
+	function __construct($request, $fileIdAndRevision = null) {
+		parent::__construct($request, $fileIdAndRevision);
+	}
+
+
+	//
+	// Implement template methods from AuthorizationPolicy
+	//
+	/**
+	 * @see AuthorizationPolicy::effect()
+	 */
+	function effect() {
+		$request = $this->getRequest();
+
+		// Get the submission file.
+		$submissionFile = $this->getSubmissionFile($request);
+		if (!is_a($submissionFile, 'SubmissionFile')) return AUTHORIZATION_DENY;
+
+		// Make sure that it's in the revision stage
+		if ($submissionFile->getFileStage() != SUBMISSION_FILE_REVIEW_REVISION) return AUTHORIZATION_DENY;
+
+		// Make sure the file belongs to the submission in request.
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		if (!is_a($submission, 'Submission')) return AUTHORIZATION_DENY;
+		if ($submission->getId() != $submissionFile->getSubmissionId()) return AUTHORIZATION_DENY;
```

Comment:
This should then also be removed from:

- SubmissionFileRequestedRevisionRequiredPolicy
- SubmissionFileSubmissionStageRequiredPolicy
- SubmissionFileViewableReviewerResponseRequiredPolicy

They are all only used in the class SubmissionFileAccessPolicy where SubmissionFileMatchesSubmissionPolicy is called at the beginning, thus they also do not need it, right?